pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -360,7 +360,7 @@ func vSphereLogin(vs *VSphere, ctx context.Context) error {
 	m := session.NewManager(vs.client.Client)
 	// retrieve client's current session
 	u, err := m.UserSession(ctx)
-	if err == nil && u == nil {
+	if err == nil && u != nil {
```

Comment:
@robdaemon Thanks for doing this. This will create user session even if it fails to obtain the existing one. So, I would suggest you to do like following,
```
. . .

if err != nil {
	glog.Errorf("Error while obtaining user session")
	return err
}

if u != nil {
	return nil
}

. . . 
```
This will handle the error. You can check conditions [here](https://github.com/vmware/govmomi/blob/7e1145231d2385c8a170c8a4a5ca11e1c1b706bb/session/manager.go#L138).